### ISSUE TITLE
SVG path rendered with offset

### ISSUE THREAD

**Issue by udifuchs** [wxWidgets/Phoenix#1957]
The following SVG:
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>                                  
<svg viewBox="0 0 2.5242536 1.0970143">                                                 
  <circle style="fill:#80c080" cx="0.5" cy="0.5" r="0.5" />                             
  <path d="m 0.5,0.5 2,-0.5" style="stroke:#008000;stroke-width:0.2" />                 
</svg> 
```
Is rendered in wx.svg like this:
![wx_test](https://user-images.githubusercontent.com/25089094/116638504-14598b80-a92c-11eb-9d42-176f0e543ffb.png)

Inkscape renders it correctly:
![wx_test_inkscape](https://user-images.githubusercontent.com/25089094/116639211-97c7ac80-a92d-11eb-8010-b668648a9768.png)

nanosvg also renders it correctly:
![wx_test_nanosvg](https://user-images.githubusercontent.com/25089094/116638896-e88ad580-a92c-11eb-891f-9a9e0c28daa8.png)

Notice that the SVG is really small. The wx.svg rendering is off by 0.5 px. It seems that somewhere px coordinates of path elements are rounded to the nearest integer.

Python 3.9.4 (default, Apr  4 2021, 19:38:44) 
[GCC 10.2.1 20210401]
wx.version: 4.1.1 gtk3 (phoenix) wxWidgets 3.1.5

**Comment by swt2c** [comment]
Unable to reproduce this - it seems like it renders correctly on my system?  This is what I see:

![Screenshot from 2021-08-23 19-48-52](https://user-images.githubusercontent.com/971544/130533831-8078abd8-338a-4186-bbbb-cb7be551f713.png)

**Comment by udifuchs** [comment]
Thanks for looking into this. I tested it again in linux (wxPython 4.1.1 and 4.1.2a1.dev5165+64e5d863) and on windows (wxPython 4.1.1) and I always get the result as in this screenshot:
![wx_svg_test_win](https://user-images.githubusercontent.com/25089094/130557484-11b98605-f792-4f23-a10e-6af13a89183c.png)

I did notice that the `SVGImage_Bitmap` demo does generate the correct output. The difference is that `SVGImage_Render` uses `RenderToGC` while `SVGImage_Bitmap` uses `ConvertToScaledBitmap`. I'll see if I can use a bitmap to workaround this bug.

Still, it does not explain why you and I are getting a different rendering of the same SVG.

**Comment by swt2c** [comment]
Would screen size / DPI make a difference?  I was testing on a HiDPI screen.

**Comment by udifuchs** [comment]
You are right about the HiDPI. I don't have a HiDPI screen. Still, once I set scaling to 200% (in gnome-control-center) the SVG renders correctly.

Printing the curve points in `wx.svg.SVGimage._makeSubPath`, I see that the values are correct and independent of my display scaling. It seems that the problem is in `wx._core.GraphicsPath`.

**Comment by ronny-rentner** [comment]
I can reproduce this problem on Debian 11 with wxpython installed and compiled through pip.

I can also confirm that using ConvertToScaledBitmap() seems to work around the problem for the givenSVG.

Though, with other SVGs, it's still broken even if you use ConvertToScaledBitmap(), e. g. I'm trying this play icon heroicons

<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
  <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M14.752 11.168l-3.197-2.132A1 1 0 0010 9.87v4.263a1 1 0 001.555.832l3.197-2.132a1 1 0 000-1.664z" />
  <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 12a9 9 0 11-18 0 9 9 0 0118 0z" />
</svg>

It should look like this (please ignore the transparent background): 
![image](https://user-images.githubusercontent.com/4419051/149479237-070b355b-4198-45bc-b686-6b840e7bda48.png)

But it wx, it loads like this: 
![image](https://user-images.githubusercontent.com/4419051/149478436-25a9551a-f8af-43b2-95af-9bdaad317cbc.png)

**Comment by ronny-rentner** [comment]
PS: nanosvg says its not maintained anymore: https://github.com/memononen/nanosvg

**Comment by ronny-rentner** [comment]
I've tried https://cairosvg.org/documentation/ using ps2png() and it renders the the svgs correctly.

**Comment by udifuchs** [comment]
The bug I was reporting is not in nanosvg, and I think that it not a wx.svg bug. It seems like a bug in wx.GraphicsRenderer and how it draws paths.

Here is test code showing the bug without using SVG:
[render_test.py.txt](https://github.com/wxWidgets/Phoenix/files/7876438/render_test.py.txt)

Which looks like this:
![image](https://user-images.githubusercontent.com/25089094/149648380-a00a25aa-e2d7-4bf2-8c1b-93fb948e2fc3.png)

It draw the circle around (0.5, 0.5) correctly. But the line that is supposed to start at (0.5, 0.5) is offset to (1, 1). You can see how it is suppose to look by changing SCALE from 1 to 100:
![image](https://user-images.githubusercontent.com/25089094/149648476-b1087853-4ecb-4d66-bfca-8c45e2726e09.png)

What seems to be happening is that shapes with a brush are drawn correctly, but shapes with a pen are somehow rounded to the next integer. It is weird, because the first point in the path is rounded to integer and the rest of the points are offset accordingly.

**Comment by udifuchs** [comment]
I rebuild wxPython with the latest wxWidgets and this bug is fixed.
It was fix in wxWidgets@52cc838b124deffa0521dbb77b901b317e70195d.
Therefore this bug should be fixed in the next release of wxPython.
